### PR TITLE
Add layout related updates to the DataForm README

### PR DESCRIPTION
**Pull request by louwie17**
## What?

Related to: https://github.com/WordPress/gutenberg/issues/59745

Updates DataViews/README.md for the `DataForm` component to reflect the new options around the layout, in particular:
- The addition of `labelPosition`
- The addition of passing an object into `form.fields` with support for `layout`, `labelPosition`
- The combined fields support, by using `children` and `label` within `form.fields`.

## Testing Instructions

1. Read over the documentation updates and see if they make sense.

### CODE REVIEW

File: packages/dataviews/README.md [hunk 1]
```diff
@@ -439,8 +441,30 @@ const fields = [
...
+			label: 'Status & Visibility',
+			children: [ 'status', 'password' ],
+		},
+		{
+			id: 'featured_media',
+			layout: 'regular',
+		},
+	],
+};
+```
```
Comment (louwie17) [inline packages/dataviews/README.md:467]: Added `labelPosition` and an example.

File: packages/dataviews/README.md [hunk 2]
```diff
@@ -1046,11 +1086,91 @@ Example:
 // Configure a filter as multi-selection with all the options.
 {
 	filterBy: {
-		operators: [ `isAny`, `isNone`, `isAll`, `isNotAll` ]
+		operators: [ `isAny`, `isNone`, `isAll`, `isNotAll` ];
 	}
 }
 ```
 
+## Form Field API
```
Comment (louwie17) [inline packages/dataviews/README.md:1094]: Added documentation around the `FormField` type, that is supported by `form.fields`

File: packages/dataviews/README.md [hunk 3]
```diff
@@ -366,11 +368,11 @@ Callback that signals the user selected one of more items. It receives the list
 
 If `selection` and `onChangeSelection` are provided, the `DataViews` component behaves like a controlled component. Otherwise, it behaves like an uncontrolled component.
 
-### `isItemClickable`: `function`
+#### `isItemClickable`: `function`
 
 A function that determines if a media field or a primary field is clickable. It receives an item as an argument and returns a boolean value indicating whether the item can be clicked.
 
-### `onClickItem`: `function`
+#### `onClickItem`: `function`
```
Comment (louwie17) [inline packages/dataviews/README.md:375]: Update the header of the above two to match the other options.

File: packages/dataviews/README.md [hunk 4]
```diff
@@ -2,8 +2,8 @@
```
Comment (louwie17) [inline packages/dataviews/README.md:2]: I left comments by the real changes, the rest are formatting done by `prettier`.

**Review by gigitux — APPROVED** [review]
Thanks for improving the documentation! LGTM!

**Review by oandregal — APPROVED** [review]
Thanks for taking care of this.

**Pull request merged**